### PR TITLE
[SPARK-30826][SQL] Respect reference case in `StringStartsWith` pushed down to parquet

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to convert the attribute name of `StringStartsWith` pushed down to the Parquet datasource to column reference via the `nameToParquetField` map. Similar conversions are performed for other source filters pushed down to parquet.

### Why are the changes needed?
This fixes the bug described in [SPARK-30826](https://issues.apache.org/jira/browse/SPARK-30826). The query from an external table:
```sql
CREATE TABLE t1 (col STRING)
USING parquet
OPTIONS (path '$path')
```
created on top of written parquet files by `Seq("42").toDF("COL").write.parquet(path)` returns wrong empty result:
```scala
spark.sql("SELECT * FROM t1 WHERE col LIKE '4%'").show
+---+
|col|
+---+
+---+
```

### Does this PR introduce any user-facing change?
Yes. After the changes the result is correct for the example above:
```scala
spark.sql("SELECT * FROM t1 WHERE col LIKE '4%'").show
+---+
|col|
+---+
| 42|
+---+
```

### How was this patch tested?
Added a test to `ParquetFilterSuite`

### CODE REVIEW

**Comment by MaxGekk** [comment]
jenkins, retest this, please

**Review by Ngone51 — APPROVED** [review]
LGTM. It'd be better if we can reduce the scope of PR title to make it more compatible with the fix.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala [hunk 1]
```diff
@@ -591,7 +591,7 @@ class ParquetFilters(
       case sources.StringStartsWith(name, prefix)
           if pushDownStartWith && canMakeFilterOn(name, prefix) =>
         Option(prefix).map { v =>
-          FilterApi.userDefined(binaryColumn(name),
+          FilterApi.userDefined(binaryColumn(nameToParquetField(name).fieldName),
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala]: good catch!

**Review by cloud-fan — APPROVED** [review]

**Review by wangyum — APPROVED** [review]

**Comment by cloud-fan** [comment]
does 2.4 also have this bug?

**Comment by MaxGekk** [comment]
@cloud-fan Yes, 2.4 has the bug too.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0/2.4!

**Review by HyukjinKwon — COMMENTED** [review]
Nice. LGTM

**Pull request closed without merging**